src/chromium/crTarget.ts

```diff
@@ -82,22 +82,26 @@ export class CRTarget {
     throw new Error('Not a page.');
   }
 
-  async serviceWorker(): Promise<Worker | null> {
-    if (this._targetInfo.type !== 'service_worker')
-      return null;
-    if (!this._workerPromise) {
-      // TODO(einbinder): Make workers send their console logs.
-      this._workerPromise = this.sessionFactory().then(session => {
-        const worker = new Worker(this._targetInfo.url);
-        session.once('Runtime.executionContextCreated', async event => {
-          worker._createExecutionContext(new CRExecutionContext(session, event.context));
-        });
-        // This might fail if the target is closed before we receive all execution contexts.
-        session.send('Runtime.enable', {}).catch(debugError);
-        return worker;
-      });
+  private async _initializeServiceWorker(session: CRSession): Promise<Worker | Error> {
+    const worker = new Worker(this._targetInfo.url);
+    session.once('Runtime.executionContextCreated', event => {
+      worker._createExecutionContext(new CRExecutionContext(session, event.context));
+    });
+    try {
+      // This might fail if the target is closed before we receive all execution contexts.
+      await session.send('Runtime.enable', {});
```

Comment:
Can we await sequentially? I was thinking this would be a `Promise.all`.

Reply:
I think so. Done.